test/tests/workspace/workspace_image_test.go

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) 2021 Gitpod GmbH. All rights reserved.
+// Licensed under the GNU Affero General Public License (AGPL).
+// See License-AGPL.txt in the project root for license information.
+
+package workspace_test
+
+import (
+	"strings"
+	"testing"
+	"time"
+
+	csapi "github.com/gitpod-io/gitpod/content-service/api"
+	imgapi "github.com/gitpod-io/gitpod/image-builder/api"
+	"github.com/gitpod-io/gitpod/test/pkg/integration"
+	agent "github.com/gitpod-io/gitpod/test/tests/workspace/workspace_agent/api"
+	"github.com/gitpod-io/gitpod/ws-manager/api"
+)
+
+func TestImageBuildPreservesEnvVarMk3(t *testing.T) {
+	it, ctx := integration.NewTest(t, 5*time.Minute)
+	defer it.Done()
+
+	res := integration.LaunchWorkspaceDirectly(it, integration.WithWorkspaceImageRequest(&imgapi.ResolveWorkspaceImageRequest{
+		Source: &imgapi.BuildSource{
+			From: &imgapi.BuildSource_File{
+				File: &imgapi.BuildSourceDockerfile{
+					DockerfileVersion: "some-version",
+					DockerfilePath:    ".gitpod.Dockerfile",
+					ContextPath:       ".",
+					Source: &csapi.WorkspaceInitializer{
+						Spec: &csapi.WorkspaceInitializer_Git{
```

Comment:
Relying on external, pre-existing fixtures is bound to break. I'd strongly vote for trying to keep these tests self-contained.

We have a content initializer which can download files from a URL. We could use GCP storage to upload the file, produce a pre-signed URL and use that to init the workspace content. Alternatively we should use GitHub API/git to set up the repo before we use it.

Reply:
> We have a content initializer which can download files from a URL.

Had the same line of thought, but ended it with "file download is equivalent to git clone". The idea with uploading the content upfront is nice, but then we should use sth that is available to all Gitpod installations (abuse content-service?). Also, we kind of break abstractions because we make assumptions about the internal structure of a workspace backup.

Alternatively we could have a variant of the `FileDownloadInitializer/a new `DirecContentInitializer` which allows to pass bytes of a `.tar.gz` directly into the initializer spec. This would be awesome for testing. WDYT? @csweichel

Reply:
I added `FilesInitializer` to see how it turns out - and it makes testing much more wholesome. :hugs: 
No hard feelings about names or impl but I like the idea.

Continuing on this: To even more encapsulate theses tests from GitHub especially we could have an CreateWorkspace API that takes an already parsed context (for testing only?). Context parsing itself should be handled by unit tests, so we don't loose anything by taking it out of the loop. Plus we achieve "full GitHub independence". :slightly_smiling_face:

Reply:
> o even more encapsulate theses tests from GitHub especially we could have an CreateWorkspace API that takes an already parsed context (for testing only?)

Thinking a bit more about it that does not make sense: We still rely on external services after context parsing (config deduction). Moving further down the chain calling `WorkspaceStarter` directly would be nice - and would eliminate the need for the "to-image-build-or-not" in the integration test suite. :grimacing: